pkg/controllers/automigration/controller.go

```diff
@@ -301,24 +301,35 @@ func (c *Controller) estimateCapacity(
 	estimatedCapacity := make(map[string]int64, len(clusterObjs))
 
 	for _, clusterObj := range clusterObjs {
-		unsObj := clusterObj.Object.(*unstructured.Unstructured)
+		keyedLogger := keyedLogger.WithValues("cluster", clusterObj.ClusterName)
+
+		unsClusterObj := clusterObj.Object.(*unstructured.Unstructured)
 
 		// NOTE: these should follow Deployment's status semantics and should not include terminating pods
-		totalReplicas, readyReplicas, err := c.getTotalAndReadyReplicas(unsObj)
+		//
+		// TODO: what if some workloads do not have a field that represents the totalReplicas or readyReplicas?
+		totalReplicas, readyReplicas, err := c.getTotalAndReadyReplicas(unsClusterObj)
 		if err != nil {
-			keyedLogger.Error(err, "Failed to get total and ready replicas from object", "cluster", clusterObj.ClusterName)
+			// TODO: since we are now using desiredReplicas - unschedulablePods to calculate estimatedCapacity,
+			// retrieving the totalReplicas and readyReplicas is now merely for optimization.
+			keyedLogger.Error(err, "Failed to get total and ready replicas from object")
 			continue
 		}
-
 		if totalReplicas == readyReplicas {
-			// no unschedulable pods
+			// no unschedulable pods, avoid unnecessary pod listing
+			continue
+		}
+
+		desiredReplicas, err := c.getDesiredReplicas(unsClusterObj)
```

Comment:
Should we add a comment to explain why we should use spec.replicas instead of status.replicas?